interp/stdarg.ml

```diff
@@ -61,9 +61,12 @@ let wit_uconstr = make0 "uconstr"
 
 let wit_open_constr = make0 ~dyn:(val_tag (topwit wit_constr)) "open_constr"
 
-let wit_clause_dft_concl  =
+let wit_clause_dft_concl =
   make0 "clause_dft_concl"
 
+let wit_binders =
```

Comment:
You should move this up into Ltac as well, in Extraargs typically.

Reply:
It is not clear to me that `binders` is essentially of different usage than `constr`. I prefer to let them together until clearer plans arrive.

Idem for `red_expr`, I prefer that a plan arrives.

Reply:
Then I am afraid that we're back to square one, insofar as it doesn't make sense to have a genarg declared there that is useless in practice, because all of the registering happens in the Ltac plugin. You can't have the genarg split in two parts that way. Either you pull everything below (which I don't like but can still be argued for) or you push everything into Ltac.

Reply:
@ppedrot, I don't know how to find a conclusion, we are turning in round. Why to want `binders` and `constr` to be only for Ltac and not for any other kind of dynamic grammar extension in general?

If your view is that `binders` and `constr` are only for Ltac what are your plans for non-Ltac grammar extensions?

Reply:
> we are turning in round

Indeed. Yet, note that ring, which is the main motivation for this PR, already depends on Ltac so it is not a problem in practice to have them live there (and also the ring PR is reinventing typeclasses, but that's another story). The only code that currently depends on this genarg is also in Ltac, further strengthening my point. I will reconsider my position if we start having non-Ltac commands that need binders in their grammar.

> what are your plans for non-Ltac grammar extensions?

Most of Genarg is only there to support Ltac-facing features. We should seriously think about splitting vernac arguments from general arguments, and move most of the Ltac-specific code up. Vernac arguments are really trivial in spirit, that's just a parsing and a printing rule, we do not even need substitution, let alone interpretation and all the fancy bells and whistles that come with having three distinct levels.